calc_all.py

```diff
@@ -217,7 +218,8 @@ def calc_checksum():
     print("#"*50)
     print("Calculating the checksum.")
     t1 = time.time()
-    datestr = "-".join([str(x) for x in time.localtime()[:3]])
+    # datestr format YYYY-MM-DD
+    datestr = "-".join([str(x).zfill(2) for x in time.localtime()[:3]])
```

Comment:
The `datestr` is used several times in the code and always re-evaluated. This should only be done once at the beginning.

Reply:
Done